examples/examples_test.go

```diff
@@ -28,120 +28,191 @@ import (
+		With(integration.ProgramTestOptions{
+			Dir: path.Join(getCwd(t), "loadbalancer"),
 			// TODO[pulumi/pulumi-terraform#241] This test currently triggers a bug in refresh, so we'll skip
 			// running the refresh step for now.
 			SkipRefresh:   true,
 			RunUpdateTest: true,
-		}),
-		jsBase.With(integration.ProgramTestOptions{
-			Dir: path.Join(cwd, "webserver"),
+		})
+
+	integration.ProgramTest(t, &test)
+}
+
+func TestAccWebserver(t *testing.T) {
+	skipIfShort(t)
+	test := getJSBaseOptions(t).
+		With(integration.ProgramTestOptions{
+			Dir: path.Join(getCwd(t), "webserver"),
 			// TODO[pulumi/pulumi-terraform#241] This test currently triggers a bug in refresh, so we'll skip
 			// running the refresh step for now.
 			SkipRefresh:   true,
 			RunUpdateTest: true,
-		}),
+		})
+
+	integration.ProgramTest(t, &test)
+}
+
+func TestAccMinimalPy(t *testing.T) {
+	test := getPythonBaseOptions(t).
+		With(integration.ProgramTestOptions{
+			Dir: path.Join(getCwd(t), "minimal-py"),
+		})
+
+	integration.ProgramTest(t, &test)
+}
+
+func TestAccBucketPy(t *testing.T) {
+	test := getPythonBaseOptions(t).
+		With(integration.ProgramTestOptions{
+			Dir:           path.Join(getCwd(t), "bucket-py"),
+			RunUpdateTest: true,
+		})
+
+	integration.ProgramTest(t, &test)
+}
+
+func TestAccDatasourcePy(t *testing.T) {
+	test := getPythonBaseOptions(t).
+		With(integration.ProgramTestOptions{
+			Dir:           path.Join(getCwd(t), "datasource-py"),
+			RunUpdateTest: true,
+		})
+
+	integration.ProgramTest(t, &test)
+}
+
+func skipIfShort(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping long-running test in short mode")
 	}
+}
 
-	tests := shortTests
-	if !testing.Short() {
-		tests = append(tests, longTests...)
+func getProject(t *testing.T) string {
+	project := os.Getenv("GOOGLE_PROJECT")
+	if project == "" {
+		t.Skipf("Skipping test due to missing GOOGLE_PROJECT environment variable")
 	}
 
-	for _, ex := range tests {
-		example := ex
-		t.Run(filepath.Base(example.Dir), func(t *testing.T) {
-			integration.ProgramTest(t, &example)
-		})
+	return project
+}
+
+func getZone(t *testing.T) string {
+	zone := os.Getenv("GOOGLE_ZONE")
+	if zone == "" {
+		t.Skipf("Skipping test due to missing GOOGLE_ZONE environment variable")
 	}
+
+	return zone
 }
 
-func createEditDir(dir string) integration.EditDir {
-	return integration.EditDir{Dir: dir, ExtraRuntimeValidation: nil}
+func getRegion(t *testing.T) string {
+	region := os.Getenv("GOOGLE_REGION")
+	if region == "" {
+		t.Skipf("Skipping test due to missing GOOGLE_REGION environment variable")
+	}
+
+	return region
+}
+
+func getCwd(t *testing.T) string {
+	cwd, err := os.Getwd()
+	if err != nil {
+		t.Error("expected a valid working directory", err)
+	}
+
+	return cwd
+}
+
+func getBaseOptions(t *testing.T) integration.ProgramTestOptions {
+	project := getProject(t)
+	zone := getZone(t)
+	region := getRegion(t)
+	return integration.ProgramTestOptions{
+		ExpectRefreshChanges: true,
```

Comment:
Similar to other PR - let’s not functionally change the tests here. Adding this non-trivially reduces the strictness of the tests - and we probably shouldn’t do that unless we need to (and for known/expected reason).

Reply:
And this is why we have reviews :) Good pick up!